File: typescript/design-patterns/observer/subject/ticker.ts
// Subject: owns the data and a list of observers.
class StockTicker {
  private observers: Observer[] = [];

  subscribe(o: Observer)   { this.observers.push(o); }
  unsubscribe(o: Observer) { this.observers = this.observers.filter(x => x !== o); }

  setPrice(p: number) {
    // The subject doesn't know WHO listens — it just broadcasts.
    for (const o of this.observers) o.update(p);
  }
}

File: go/design-patterns/observer/subject/ticker.go
type StockTicker struct {
    observers []Observer
}

func (t *StockTicker) Subscribe(o Observer) {
    t.observers = append(t.observers, o)
}

func (t *StockTicker) Unsubscribe(target Observer) {
    kept := t.observers[:0]
    for _, o := range t.observers {
        if o != target {
            kept = append(kept, o)
        }
    }
    t.observers = kept
}

func (t *StockTicker) SetPrice(p float64) {
    // The subject doesn't know WHO listens — it just broadcasts.
    for _, o := range t.observers {
        o.Update(p)
    }
}